x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -190,6 +191,17 @@ protected void masterOperation(
                 return;
             }
 
+            if (licenseState.isAllowedByLicense(config.getInferenceConfig().getMinLicenseSupported()) == false) {
```

Comment:
There is already a licence check at inference time but that is for the lower level platinum licence. 

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java#L116


The code allows for a per model licence check but the mechanism is different, it uses the `TrainedModelConfig::getLicenseLevel()` function. Licence level cannot be set by the user, for internal models it is hard coded to the correct level

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java#L976

Reply:
👋  Thanks @davidkyle - I'll be taking over the PR from @afoucret at the moment... 

Just so I'm on the same page with what you're saying, are you saying that we don't need to have this entire `if (licenseState.isAllowedByLicense...` block here because the `getMinLicenseSupported()` check happens elsewhere before this? Thanks!

Reply:
@davidkyle - I've been looking at this code [you linked to above](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java#L116) and I _think_ I'm figuring this out... 

This is only for the license check at inference time, correct? So, then the `getLicenseLevel()` methods needs to be overridden in the LTR config to ensure it returns `BASIC` as we don't want to stop the users from using the model if they don't have a Plat+ license - but, we still want to only allow Enterprise licensed users to _create_ the inference model, which is why I think @afoucret has that check added to the `TransportPutTrainModelAction` class... trying to think of how best to ensure we can have two different levels for create vs. infer... maybe I need to add a new base method akin to `getLicenseLevel()` but explicitly for creating a model (PUT)?